### PR TITLE
Values bind join

### PR DESCRIPTION
**Pull request by scferrada**
Tests aren't passing because of a ``ClassCastException``, where the test suite tries to cast a ``SPARQLGraphPattern`` into a ``TriplePattern`` on the ``performRequest`` method. Not sure if that is the expected behaviour.

### CODE REVIEW

File: src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpVALUESBindJoin.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import se.liu.ida.hefquin.engine.federation.SPARQLEndpoint;
+import se.liu.ida.hefquin.engine.queryplan.ExpectedVariables;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.ExecOpBindJoinSPARQLwithVALUES;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.UnaryExecutableOp;
+import se.liu.ida.hefquin.engine.queryplan.logical.UnaryLogicalOp;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpTPAdd;
+
+public class PhysicalOpVALUESBindJoin extends BasePhysicalOpSingleInputJoin {
+
+	protected PhysicalOpVALUESBindJoin(UnaryLogicalOp lop) {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpVALUESBindJoin.java:13]:
```suggestion
	public PhysicalOpVALUESBindJoin( final UnaryLogicalOp lop ) {
```

File: src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpVALUESBindJoin.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		super(lop);
+	}
+
+	@Override
+	public UnaryExecutableOp createExecOp(ExpectedVariables... inputVars) {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpVALUESBindJoin.java:18]:
```suggestion
	public UnaryExecutableOp createExecOp( final ExpectedVariables... inputVars ) {
```

**Comment by hartig** [comment]
This is for #32

File: src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpVALUESBindJoin.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import se.liu.ida.hefquin.engine.federation.FederationMember;
+import se.liu.ida.hefquin.engine.federation.SPARQLEndpoint;
+import se.liu.ida.hefquin.engine.queryplan.ExpectedVariables;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.ExecOpBindJoinSPARQLwithVALUES;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.UnaryExecutableOp;
+import se.liu.ida.hefquin.engine.queryplan.logical.UnaryLogicalOp;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpTPAdd;
+
+public class PhysicalOpVALUESBindJoin extends BasePhysicalOpSingleInputJoin {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpVALUESBindJoin.java:11]: Please rename this class to `PhysicalOpBindJoinWithVALUES`.

File: src/test/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithVALUESTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		_tpWithEmptySolutionMappingAsInput();
+	}
+
+	@Test
+	public void tpWithEmptyResponses() {
+		_tpWithEmptyResponses();
+	}
+
+	@Override
+	protected SPARQLEndpoint createFedMemberForTest(Graph dataForMember) {
```
Comment (hartig) [inline src/test/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithVALUESTest.java:42]:
```suggestion
	protected SPARQLEndpoint createFedMemberForTest( final Graph dataForMember ) {
```

File: src/test/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithVALUESTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+		return new SPARQLEndpointForTest(dataForMember);
+	}
+
+	@Override
+	protected UnaryExecutableOp createExecOpForTest(TriplePattern tp, SPARQLEndpoint fm) {
```
Comment (hartig) [inline src/test/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpBindJoinSPARQLwithVALUESTest.java:47]:
```suggestion
	protected UnaryExecutableOp createExecOpForTest( final TriplePattern tp, final SPARQLEndpoint fm ) {
```

**Review by hartig — CHANGES_REQUESTED** [review]

**Comment by hartig** [comment]
> Tests aren't passing because of a `ClassCastException`, where the test suite tries to cast a `SPARQLGraphPattern` into a `TriplePattern` on the `performRequest` method.

I am afraid the test suite (`EngineTestBase`) needs to be extended to be able to deal with cases in which the given `SPARQLRequest` is not a `TriplePatternRequest` and its query pattern is not a `TriplePattern` but just a `SPARQLGraphPattern` with an `Op` object. For such a case, there should be a way to have Jena execute that `Op` operator over the given `Graph`. Can you please add this.

**Review by hartig — CHANGES_REQUESTED** [review]

**Comment by hartig** [comment]
...plus the failing unit test.

**Comment by scferrada** [comment]
All tests pass now

**Pull request merged**